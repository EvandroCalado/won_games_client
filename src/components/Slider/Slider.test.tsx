import { render, screen } from '@testing-library/react';
import { Slider } from '.';

describe('Slider', () => {
  it('should render children as slider item', () => {
    render(
      <Slider settings={{ slidesToShow: 1, infinite: false }}>
        <p>item1</p>
        <p>item2</p>
      </Slider>,
    );

    expect(screen.getByText(/item1/i).parentElement?.parentElement).toHaveClass(
      'slick-slide',
    );
    expect(screen.getByText(/item2/i).parentElement?.parentElement).toHaveClass(
      'slick-slide',
    );
  });

  it('should render a snapshot', () => {
    const { container } = render(
      <Slider settings={{ slidesToShow: 1, infinite: false }}>
        <p>item1</p>
        <p>item2</p>
      </Slider>,
    );

    expect(container).toMatchSnapshot();
  });
});
