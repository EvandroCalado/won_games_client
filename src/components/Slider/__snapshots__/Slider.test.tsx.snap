// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render a snapshot 1`] = `
<div>
  <div
    class="slick-slider slick-initialized"
    dir="ltr"
  >
    <button
      class="slick-arrow slick-prev slick-disabled"
      data-role="none"
      style="display: block;"
      type="button"
    >
       
      Previous
    </button>
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <p
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              item1
            </p>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="1"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <p
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              item2
            </p>
          </div>
        </div>
      </div>
    </div>
    <button
      class="slick-arrow slick-next"
      data-role="none"
      style="display: block;"
      type="button"
    >
       
      Next
    </button>
  </div>
</div>
`;
